packages/repl/src/index.js

```diff
@@ -278,9 +281,20 @@ export function createRepl({ isVerbose, onAssert, onLog, onOk }) {
     const importObject = {
       _internalInstanceOptions: internalInstanceOptions
     };
-    const module = createCompiledModule(moduleNode);
 
-    return new Instance(module, importObject);
+    if (enableTypeChecking === true) {
+      denormalizeTypeReferences(moduleNode);
+
+      const typeErrors = typeCheck(t.program([moduleNode]));
```

Comment:
I just noticed that but type checking should be done an a module directly.

- Program can have multiple modules
- Program have no typing rules
- in fact I created the notion of Program